### ISSUE TITLE
False positive match for URL route definitions

### ISSUE THREAD

**Issue by bennidi** [chrmarti/vscode-regex#27]
Hey, I really enjoy this small extension a lot. Thanks for giving it out to the community.
I am currently developing a lot of node.js / koa services. And the codebase also contains quite some swagger documentation.

I have matches in various places where I don't like them and all the extra lenses make me see less than more :)
Example:

```js
    describe('(a) POST /vehicles/{vehicleId}/reservations', () => {
```

I don't know what other users think but I believe that test regex should never be applied to strings that have some regex like meta-character in them.

**Comment by chrmarti** [comment]
I agree. The difficulty here is that the extension currently doesn't know which part of the code it is looking at. So it doesn't know if it is in a string, comment or regular code.

**Comment by bennidi** [comment]
I have no clue how vscode extensions are developed -  I mean what the
surrounding framework provides as context and hooks etc. But in an IDE,
shouldn't there be an AST of some sort? Anyways, the regex should be able
to see that it is contained in a string (^' or ^" and $' , $") - but I am
also no regexpert.

What do you think? What would be a good way to go? Maybe I can create a
PR...

2018-04-20 11:43 GMT+02:00 Christof Marti <notifications@github.com>:

**Comment by chrmarti** [comment]
There is an AST, but that is in a different extension where this extension does not have access. While somewhat limiting in general, it also avoids the large API exposure that an AST would carry.

This extension could try to do its own scanning of strings, comments, etc., but that would partly reinvent the parser and might be hard to get right.
